app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt

```diff
@@ -118,6 +121,40 @@ class SkillDetailsFragment : Fragment(), ISkillDetailsView {
         }
     }
 
+    private fun setReportButton() {
+        //Only enable the feature when the user is logged in
+        if (PrefManager.getToken() != null) {
+            reportSkill.visibility = View.VISIBLE
+            reportSkill.setOnClickListener {
+                val dialogBuilder = AlertDialog.Builder(requireContext())
+                val view = layoutInflater.inflate(R.layout.alert_report_skill, null)
+                val reportMsgText = view.findViewById(R.id.reportMsg) as EditText
+                dialogBuilder.setView(view)
+                dialogBuilder.setTitle(R.string.report_skill)
+
+                dialogBuilder.setPositiveButton(R.string.report_send) { dialog, whichButton ->
+                    if (PrefManager.getToken() != null && reportMsgText.text.isNotEmpty()) {
+                        val map: MutableMap<String, String> = HashMap()
+                        map.put("access_token", PrefManager.getToken().toString())
+                        map.put("feedback", reportMsgText.text.toString())
+                        map.put("skill", skillTag)
+                        map.put("group", skillData.group)
+                        map.put("model", skillData.model)
+                        skillDetailsPresenter.sendReport(map)
```

Comment:
Use objects.

Reply:
@arundhati24 Couldn't get it?Please tell me more